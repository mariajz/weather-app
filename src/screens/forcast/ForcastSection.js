import React from 'react';
import {
    fetchWeatherImage,
    filterTimeFromTimestamp,
} from '../../utils/helpers';
import { ForCastSectionWrapper } from './ForcastSection.style';
import DailyForcastScroll from '../../components/daily-forcast-scroll';
import WeatherDetailsSection from '../../components/weather-details-section';
import { mockSuccessResponse } from '../../api/weather-api/current-weather/mocks';

const ForcastSection = () => {
    const data = mockSuccessResponse;
    const locationProps = {
        locationName: data.location.name,
        region: data.location.region,
        country: data.location.country,
    };
    const weatherIconProps = { weatherType: data.current.condition.text };
    const weatherDetailsProps = {
        temperature: data.current.temp_c,
        condition: data.current.condition.text,
    };
    const weatherSummaryProps = {
        wind: data.current.wind_kph,
        uv: data.current.uv,
        humidity: data.current.humidity,
    };

    const fetchTodaysForcastData = forecastData =>
        forecastData.map(item => ({
            temp: item.temp_c,
            condition: item.condition.text,
            time: filterTimeFromTimestamp(item.time),
            weatherImage: fetchWeatherImage(item.condition.text),
        }));

    const dailyForcastScrollProps = {
        forecastData: fetchTodaysForcastData(data.forecast.forecastday[0].hour),
    };

    return (
        <>
            <ForCastSectionWrapper>
                <WeatherDetailsSection
                    locationProps={locationProps}
                    weatherIconProps={weatherIconProps}
                    weatherDetailsProps={weatherDetailsProps}
                    weatherSummaryProps={weatherSummaryProps}
                    dailyForcastScrollProps={dailyForcastScrollProps}
                />
                <DailyForcastScroll
                    dailyForcastScrollProps={dailyForcastScrollProps}
                />
            </ForCastSectionWrapper>
        </>
    );
};

export default ForcastSection;
