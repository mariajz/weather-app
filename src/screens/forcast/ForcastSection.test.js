import { render } from '@testing-library/react-native';
import React from 'react';
import ForcastSection from './ForcastSection';

describe('ForcastSection', () => {
    it('should render ForcastSection', () => {
        const container = render(<ForcastSection />);

        expect(container).toMatchSnapshot();
    });
});
