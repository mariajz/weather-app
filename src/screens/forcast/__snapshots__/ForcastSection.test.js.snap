// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForcastSection should render ForcastSection 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "column",
      "height": "90%",
      "paddingBottom": 8,
      "paddingLeft": 8,
      "paddingRight": 8,
      "paddingTop": 8,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "height": "60%",
        "justifyContent": "space-between",
        "paddingBottom": 15,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 15,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 30,
            "fontWeight": "bold",
          }
        }
      >
        Manhatten
      </Text>
      <Text
        style={
          {
            "color": "white",
            "fontSize": 15,
          }
        }
      >
        New York
        , 
        United States of America
      </Text>
    </View>
    <Image
      height={100}
      source={
        {
          "testUri": "../../../assets/images/sun.png",
        }
      }
      style={
        {
          "height": 100,
          "width": 100,
        }
      }
      testID="image"
      width={100}
    />
    <View
      style={
        {
          "alignItems": "center",
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 30,
            "fontWeight": "bold",
          }
        }
      >
        27.8
        Â°
        C
      </Text>
      <Text
        size={22}
        style={
          {
            "color": "white",
            "fontSize": 22,
            "marginLeft": 0,
          }
        }
      >
        Sunny
      </Text>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        style={
          {
            "flexDirection": "row",
            "paddingBottom": 10,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 10,
          }
        }
      >
        <Image
          source={
            {
              "testUri": "../../../assets/icons/wind.png",
            }
          }
          style={
            {
              "height": 25,
              "width": 25,
            }
          }
          testID="icon"
          tintColor="white"
        />
        <Text
          marginLeft={5}
          size={18}
          style={
            {
              "color": "white",
              "fontSize": 18,
              "marginLeft": 5,
            }
          }
        >
          3.6
           km
        </Text>
      </View>
      <View
        style={
          {
            "flexDirection": "row",
            "paddingBottom": 10,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 10,
          }
        }
      >
        <Image
          source={
            {
              "testUri": "../../../assets/icons/humidity.png",
            }
          }
          style={
            {
              "height": 25,
              "width": 25,
            }
          }
          testID="icon"
          tintColor="white"
        />
        <Text
          marginLeft={5}
          size={18}
          style={
            {
              "color": "white",
              "fontSize": 18,
              "marginLeft": 5,
            }
          }
        >
          51
          %
        </Text>
      </View>
      <View
        style={
          {
            "flexDirection": "row",
            "paddingBottom": 10,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 10,
          }
        }
      >
        <Image
          source={
            {
              "testUri": "../../../assets/icons/sun.png",
            }
          }
          style={
            {
              "height": 25,
              "width": 25,
            }
          }
          testID="icon"
          tintColor="white"
        />
        <Text
          marginLeft={5}
          size={18}
          style={
            {
              "color": "white",
              "fontSize": 18,
              "marginLeft": 5,
            }
          }
        >
          6
           UV
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      {
        "height": "35%",
        "paddingBottom": 10,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 10,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 10,
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../assets/icons/calendar.png",
          }
        }
        style={
          {
            "height": 25,
            "width": 25,
          }
        }
        testID="icon"
        tintColor="white"
      />
      <Text
        marginLeft={5}
        size={14}
        style={
          {
            "color": "white",
            "fontSize": 14,
            "marginLeft": 5,
          }
        }
      >
        Today's forcast
      </Text>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "width": "100%",
        }
      }
    >
      <RCTScrollView
        data={
          [
            {
              "condition": "Clear",
              "temp": 24.9,
              "time": "12 AM",
              "weatherImage": {
                "testUri": "../../../assets/images/sun.png",
              },
            },
            {
              "condition": "Clear",
              "temp": 24.4,
              "time": "1 AM",
              "weatherImage": {
                "testUri": "../../../assets/images/sun.png",
              },
            },
            {
              "condition": "Clear",
              "temp": 24,
              "time": "2 AM",
              "weatherImage": {
                "testUri": "../../../assets/images/sun.png",
              },
            },
            {
              "condition": "Partly cloudy",
              "temp": 23.5,
              "time": "3 AM",
              "weatherImage": {
                "testUri": "../../../assets/images/partlycloudy.png",
              },
            },
            {
              "condition": "Partly cloudy",
              "temp": 23.1,
              "time": "4 AM",
              "weatherImage": {
                "testUri": "../../../assets/images/partlycloudy.png",
              },
            },
            {
              "condition": "Clear",
              "temp": 22.4,
              "time": "5 AM",
              "weatherImage": {
                "testUri": "../../../assets/images/sun.png",
              },
            },
            {
              "condition": "Cloudy",
              "temp": 21.6,
              "time": "6 AM",
              "weatherImage": {
                "testUri": "../../../assets/images/cloud.png",
              },
            },
            {
              "condition": "Cloudy",
              "temp": 21.1,
              "time": "7 AM",
              "weatherImage": {
                "testUri": "../../../assets/images/cloud.png",
              },
            },
            {
              "condition": "Sunny",
              "temp": 22.2,
              "time": "8 AM",
              "weatherImage": {
                "testUri": "../../../assets/images/sun.png",
              },
            },
            {
              "condition": "Overcast",
              "temp": 25.8,
              "time": "9 AM",
              "weatherImage": {
                "testUri": "../../../assets/images/cloud.png",
              },
            },
            {
              "condition": "Overcast",
              "temp": 26.3,
              "time": "10 AM",
              "weatherImage": {
                "testUri": "../../../assets/images/cloud.png",
              },
            },
            {
              "condition": "Sunny",
              "temp": 28.1,
              "time": "11 AM",
              "weatherImage": {
                "testUri": "../../../assets/images/sun.png",
              },
            },
            {
              "condition": "Cloudy",
              "temp": 29.8,
              "time": "12 PM",
              "weatherImage": {
                "testUri": "../../../assets/images/cloud.png",
              },
            },
            {
              "condition": "Cloudy",
              "temp": 30.8,
              "time": "1 PM",
              "weatherImage": {
                "testUri": "../../../assets/images/cloud.png",
              },
            },
            {
              "condition": "Overcast",
              "temp": 29,
              "time": "2 PM",
              "weatherImage": {
                "testUri": "../../../assets/images/cloud.png",
              },
            },
            {
              "condition": "Overcast",
              "temp": 30.7,
              "time": "3 PM",
              "weatherImage": {
                "testUri": "../../../assets/images/cloud.png",
              },
            },
            {
              "condition": "Overcast",
              "temp": 30.5,
              "time": "4 PM",
              "weatherImage": {
                "testUri": "../../../assets/images/cloud.png",
              },
            },
            {
              "condition": "Overcast",
              "temp": 25.9,
              "time": "5 PM",
              "weatherImage": {
                "testUri": "../../../assets/images/cloud.png",
              },
            },
            {
              "condition": "Overcast",
              "temp": 28.7,
              "time": "6 PM",
              "weatherImage": {
                "testUri": "../../../assets/images/cloud.png",
              },
            },
            {
              "condition": "Overcast",
              "temp": 27.4,
              "time": "7 PM",
              "weatherImage": {
                "testUri": "../../../assets/images/cloud.png",
              },
            },
            {
              "condition": "Overcast",
              "temp": 23.8,
              "time": "8 PM",
              "weatherImage": {
                "testUri": "../../../assets/images/cloud.png",
              },
            },
            {
              "condition": "Overcast",
              "temp": 25.5,
              "time": "9 PM",
              "weatherImage": {
                "testUri": "../../../assets/images/cloud.png",
              },
            },
            {
              "condition": "Overcast",
              "temp": 25,
              "time": "10 PM",
              "weatherImage": {
                "testUri": "../../../assets/images/cloud.png",
              },
            },
            {
              "condition": "Overcast",
              "temp": 23.3,
              "time": "11 PM",
              "weatherImage": {
                "testUri": "../../../assets/images/cloud.png",
              },
            },
          ]
        }
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={true}
        keyExtractor={[Function]}
        onContentSizeChange={[Function]}
        onLayout={[Function]}
        onMomentumScrollBegin={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        showsHorizontalScrollIndicator={false}
        stickyHeaderIndices={[]}
        style={
          {
            "overflow": "visible",
          }
        }
        viewabilityConfigCallbackPairs={[]}
      >
        <View>
          <View
            onFocusCapture={[Function]}
            onLayout={[Function]}
            style={
              [
                {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "rgba(255,255,255, 0.2)",
                  "borderBottomLeftRadius": 5,
                  "borderBottomRightRadius": 5,
                  "borderTopLeftRadius": 5,
                  "borderTopRightRadius": 5,
                  "height": 150,
                  "justifyContent": "space-between",
                  "marginRight": 10,
                  "paddingBottom": 10,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                  "paddingTop": 10,
                  "width": 100,
                }
              }
            >
              <Text
                size={14}
                style={
                  {
                    "color": "white",
                    "fontSize": 14,
                    "marginLeft": 0,
                  }
                }
              >
                12 AM
              </Text>
              <Image
                height={40}
                source={
                  {
                    "testUri": "../../../assets/images/sun.png",
                  }
                }
                style={
                  {
                    "height": 40,
                    "width": 40,
                  }
                }
                testID="image"
                width={40}
              />
              <Text
                size={14}
                style={
                  {
                    "color": "white",
                    "fontSize": 14,
                    "marginLeft": 0,
                  }
                }
              >
                Clear
              </Text>
              <Text
                size={12}
                style={
                  {
                    "color": "white",
                    "fontSize": 12,
                    "marginLeft": 0,
                  }
                }
              >
                24.9
                Â°
                C
              </Text>
            </View>
          </View>
          <View
            onFocusCapture={[Function]}
            onLayout={[Function]}
            style={
              [
                {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "rgba(255,255,255, 0.2)",
                  "borderBottomLeftRadius": 5,
                  "borderBottomRightRadius": 5,
                  "borderTopLeftRadius": 5,
                  "borderTopRightRadius": 5,
                  "height": 150,
                  "justifyContent": "space-between",
                  "marginRight": 10,
                  "paddingBottom": 10,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                  "paddingTop": 10,
                  "width": 100,
                }
              }
            >
              <Text
                size={14}
                style={
                  {
                    "color": "white",
                    "fontSize": 14,
                    "marginLeft": 0,
                  }
                }
              >
                1 AM
              </Text>
              <Image
                height={40}
                source={
                  {
                    "testUri": "../../../assets/images/sun.png",
                  }
                }
                style={
                  {
                    "height": 40,
                    "width": 40,
                  }
                }
                testID="image"
                width={40}
              />
              <Text
                size={14}
                style={
                  {
                    "color": "white",
                    "fontSize": 14,
                    "marginLeft": 0,
                  }
                }
              >
                Clear
              </Text>
              <Text
                size={12}
                style={
                  {
                    "color": "white",
                    "fontSize": 12,
                    "marginLeft": 0,
                  }
                }
              >
                24.4
                Â°
                C
              </Text>
            </View>
          </View>
          <View
            onFocusCapture={[Function]}
            onLayout={[Function]}
            style={
              [
                {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "rgba(255,255,255, 0.2)",
                  "borderBottomLeftRadius": 5,
                  "borderBottomRightRadius": 5,
                  "borderTopLeftRadius": 5,
                  "borderTopRightRadius": 5,
                  "height": 150,
                  "justifyContent": "space-between",
                  "marginRight": 10,
                  "paddingBottom": 10,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                  "paddingTop": 10,
                  "width": 100,
                }
              }
            >
              <Text
                size={14}
                style={
                  {
                    "color": "white",
                    "fontSize": 14,
                    "marginLeft": 0,
                  }
                }
              >
                2 AM
              </Text>
              <Image
                height={40}
                source={
                  {
                    "testUri": "../../../assets/images/sun.png",
                  }
                }
                style={
                  {
                    "height": 40,
                    "width": 40,
                  }
                }
                testID="image"
                width={40}
              />
              <Text
                size={14}
                style={
                  {
                    "color": "white",
                    "fontSize": 14,
                    "marginLeft": 0,
                  }
                }
              >
                Clear
              </Text>
              <Text
                size={12}
                style={
                  {
                    "color": "white",
                    "fontSize": 12,
                    "marginLeft": 0,
                  }
                }
              >
                24
                Â°
                C
              </Text>
            </View>
          </View>
          <View
            onFocusCapture={[Function]}
            onLayout={[Function]}
            style={
              [
                {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "rgba(255,255,255, 0.2)",
                  "borderBottomLeftRadius": 5,
                  "borderBottomRightRadius": 5,
                  "borderTopLeftRadius": 5,
                  "borderTopRightRadius": 5,
                  "height": 150,
                  "justifyContent": "space-between",
                  "marginRight": 10,
                  "paddingBottom": 10,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                  "paddingTop": 10,
                  "width": 100,
                }
              }
            >
              <Text
                size={14}
                style={
                  {
                    "color": "white",
                    "fontSize": 14,
                    "marginLeft": 0,
                  }
                }
              >
                3 AM
              </Text>
              <Image
                height={40}
                source={
                  {
                    "testUri": "../../../assets/images/partlycloudy.png",
                  }
                }
                style={
                  {
                    "height": 40,
                    "width": 40,
                  }
                }
                testID="image"
                width={40}
              />
              <Text
                size={14}
                style={
                  {
                    "color": "white",
                    "fontSize": 14,
                    "marginLeft": 0,
                  }
                }
              >
                Partly cloudy
              </Text>
              <Text
                size={12}
                style={
                  {
                    "color": "white",
                    "fontSize": 12,
                    "marginLeft": 0,
                  }
                }
              >
                23.5
                Â°
                C
              </Text>
            </View>
          </View>
          <View
            onFocusCapture={[Function]}
            onLayout={[Function]}
            style={
              [
                {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "rgba(255,255,255, 0.2)",
                  "borderBottomLeftRadius": 5,
                  "borderBottomRightRadius": 5,
                  "borderTopLeftRadius": 5,
                  "borderTopRightRadius": 5,
                  "height": 150,
                  "justifyContent": "space-between",
                  "marginRight": 10,
                  "paddingBottom": 10,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                  "paddingTop": 10,
                  "width": 100,
                }
              }
            >
              <Text
                size={14}
                style={
                  {
                    "color": "white",
                    "fontSize": 14,
                    "marginLeft": 0,
                  }
                }
              >
                4 AM
              </Text>
              <Image
                height={40}
                source={
                  {
                    "testUri": "../../../assets/images/partlycloudy.png",
                  }
                }
                style={
                  {
                    "height": 40,
                    "width": 40,
                  }
                }
                testID="image"
                width={40}
              />
              <Text
                size={14}
                style={
                  {
                    "color": "white",
                    "fontSize": 14,
                    "marginLeft": 0,
                  }
                }
              >
                Partly cloudy
              </Text>
              <Text
                size={12}
                style={
                  {
                    "color": "white",
                    "fontSize": 12,
                    "marginLeft": 0,
                  }
                }
              >
                23.1
                Â°
                C
              </Text>
            </View>
          </View>
          <View
            onFocusCapture={[Function]}
            onLayout={[Function]}
            style={
              [
                {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "rgba(255,255,255, 0.2)",
                  "borderBottomLeftRadius": 5,
                  "borderBottomRightRadius": 5,
                  "borderTopLeftRadius": 5,
                  "borderTopRightRadius": 5,
                  "height": 150,
                  "justifyContent": "space-between",
                  "marginRight": 10,
                  "paddingBottom": 10,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                  "paddingTop": 10,
                  "width": 100,
                }
              }
            >
              <Text
                size={14}
                style={
                  {
                    "color": "white",
                    "fontSize": 14,
                    "marginLeft": 0,
                  }
                }
              >
                5 AM
              </Text>
              <Image
                height={40}
                source={
                  {
                    "testUri": "../../../assets/images/sun.png",
                  }
                }
                style={
                  {
                    "height": 40,
                    "width": 40,
                  }
                }
                testID="image"
                width={40}
              />
              <Text
                size={14}
                style={
                  {
                    "color": "white",
                    "fontSize": 14,
                    "marginLeft": 0,
                  }
                }
              >
                Clear
              </Text>
              <Text
                size={12}
                style={
                  {
                    "color": "white",
                    "fontSize": 12,
                    "marginLeft": 0,
                  }
                }
              >
                22.4
                Â°
                C
              </Text>
            </View>
          </View>
          <View
            onFocusCapture={[Function]}
            onLayout={[Function]}
            style={
              [
                {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "rgba(255,255,255, 0.2)",
                  "borderBottomLeftRadius": 5,
                  "borderBottomRightRadius": 5,
                  "borderTopLeftRadius": 5,
                  "borderTopRightRadius": 5,
                  "height": 150,
                  "justifyContent": "space-between",
                  "marginRight": 10,
                  "paddingBottom": 10,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                  "paddingTop": 10,
                  "width": 100,
                }
              }
            >
              <Text
                size={14}
                style={
                  {
                    "color": "white",
                    "fontSize": 14,
                    "marginLeft": 0,
                  }
                }
              >
                6 AM
              </Text>
              <Image
                height={40}
                source={
                  {
                    "testUri": "../../../assets/images/cloud.png",
                  }
                }
                style={
                  {
                    "height": 40,
                    "width": 40,
                  }
                }
                testID="image"
                width={40}
              />
              <Text
                size={14}
                style={
                  {
                    "color": "white",
                    "fontSize": 14,
                    "marginLeft": 0,
                  }
                }
              >
                Cloudy
              </Text>
              <Text
                size={12}
                style={
                  {
                    "color": "white",
                    "fontSize": 12,
                    "marginLeft": 0,
                  }
                }
              >
                21.6
                Â°
                C
              </Text>
            </View>
          </View>
          <View
            onFocusCapture={[Function]}
            onLayout={[Function]}
            style={
              [
                {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "rgba(255,255,255, 0.2)",
                  "borderBottomLeftRadius": 5,
                  "borderBottomRightRadius": 5,
                  "borderTopLeftRadius": 5,
                  "borderTopRightRadius": 5,
                  "height": 150,
                  "justifyContent": "space-between",
                  "marginRight": 10,
                  "paddingBottom": 10,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                  "paddingTop": 10,
                  "width": 100,
                }
              }
            >
              <Text
                size={14}
                style={
                  {
                    "color": "white",
                    "fontSize": 14,
                    "marginLeft": 0,
                  }
                }
              >
                7 AM
              </Text>
              <Image
                height={40}
                source={
                  {
                    "testUri": "../../../assets/images/cloud.png",
                  }
                }
                style={
                  {
                    "height": 40,
                    "width": 40,
                  }
                }
                testID="image"
                width={40}
              />
              <Text
                size={14}
                style={
                  {
                    "color": "white",
                    "fontSize": 14,
                    "marginLeft": 0,
                  }
                }
              >
                Cloudy
              </Text>
              <Text
                size={12}
                style={
                  {
                    "color": "white",
                    "fontSize": 12,
                    "marginLeft": 0,
                  }
                }
              >
                21.1
                Â°
                C
              </Text>
            </View>
          </View>
          <View
            onFocusCapture={[Function]}
            onLayout={[Function]}
            style={
              [
                {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "rgba(255,255,255, 0.2)",
                  "borderBottomLeftRadius": 5,
                  "borderBottomRightRadius": 5,
                  "borderTopLeftRadius": 5,
                  "borderTopRightRadius": 5,
                  "height": 150,
                  "justifyContent": "space-between",
                  "marginRight": 10,
                  "paddingBottom": 10,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                  "paddingTop": 10,
                  "width": 100,
                }
              }
            >
              <Text
                size={14}
                style={
                  {
                    "color": "white",
                    "fontSize": 14,
                    "marginLeft": 0,
                  }
                }
              >
                8 AM
              </Text>
              <Image
                height={40}
                source={
                  {
                    "testUri": "../../../assets/images/sun.png",
                  }
                }
                style={
                  {
                    "height": 40,
                    "width": 40,
                  }
                }
                testID="image"
                width={40}
              />
              <Text
                size={14}
                style={
                  {
                    "color": "white",
                    "fontSize": 14,
                    "marginLeft": 0,
                  }
                }
              >
                Sunny
              </Text>
              <Text
                size={12}
                style={
                  {
                    "color": "white",
                    "fontSize": 12,
                    "marginLeft": 0,
                  }
                }
              >
                22.2
                Â°
                C
              </Text>
            </View>
          </View>
          <View
            onFocusCapture={[Function]}
            onLayout={[Function]}
            style={
              [
                {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "rgba(255,255,255, 0.2)",
                  "borderBottomLeftRadius": 5,
                  "borderBottomRightRadius": 5,
                  "borderTopLeftRadius": 5,
                  "borderTopRightRadius": 5,
                  "height": 150,
                  "justifyContent": "space-between",
                  "marginRight": 10,
                  "paddingBottom": 10,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                  "paddingTop": 10,
                  "width": 100,
                }
              }
            >
              <Text
                size={14}
                style={
                  {
                    "color": "white",
                    "fontSize": 14,
                    "marginLeft": 0,
                  }
                }
              >
                9 AM
              </Text>
              <Image
                height={40}
                source={
                  {
                    "testUri": "../../../assets/images/cloud.png",
                  }
                }
                style={
                  {
                    "height": 40,
                    "width": 40,
                  }
                }
                testID="image"
                width={40}
              />
              <Text
                size={14}
                style={
                  {
                    "color": "white",
                    "fontSize": 14,
                    "marginLeft": 0,
                  }
                }
              >
                Overcast
              </Text>
              <Text
                size={12}
                style={
                  {
                    "color": "white",
                    "fontSize": 12,
                    "marginLeft": 0,
                  }
                }
              >
                25.8
                Â°
                C
              </Text>
            </View>
          </View>
          <View
            style={
              {
                "width": 0,
              }
            }
          />
        </View>
      </RCTScrollView>
    </View>
  </View>
</View>
`;
